consensus/polybft/consensus_runtime.go

```diff
@@ -1144,7 +1151,26 @@ func (c *consensusRuntime) HasQuorum(
 			return c.fsm.validators.HasQuorum(signers)
 		}
 
-		return c.fsm.validators.HasQuorumWithoutProposer(signers)
+		if len(messages) == 0 {
+			return false
+		}
+
+		propAddress, exist := c.fsm.proposerCalculator.GetLatestProposer(messages[0].View.Round)
```

Comment:
What is this?

Reply:
Oh yea. There was a discussion about that not so long ago. There is a situation when go-ibft calls backend  and asks for quorum, but that quorum must include proposer even though a message form the proposer hasn't been submitted to the backend. Therefore, we have to find the proposer by ourselves and include it in the quorum calculation.

Reply:
here, we do not need to calculate again proposer, we just need to retrieve previously calculated proposer for specific round